### PR TITLE
fix!: use Double Cookie Submission for Fusion CSRF protection

### LINKED ISSUE #10703: Use Double Submit Cookie for Fusion CSRF token
Instead of storing the CSRF token in the session, the [Double Submit Cookie](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#double-submit-cookie) approach should be used.

The CSRF token would be put in a cookie, and also be sent in a header as is currently done. The server compares that these two tokens are equal.

This has the benefit of enabling CSRF even when there is no session, for example for stateless applications.

### PR DESCRIPTION
**Pull request by platosha**
Fixes #10703

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/VaadinServlet.java [hunk 1]
```diff
@@ -561,4 +565,33 @@ private VaadinServletContext initializeContext() {
...
+        }
+
+        /*
+         * Despite section 6 of RFC 4122, this particular use of UUID *is*
+         * adequate for security capabilities. Type 4 UUIDs contain 122 bits of
+         * random data, and UUID.randomUUID() is defined to use a
+         * cryptographically secure random generator.
+         */
+        final String csrfToken = UUID.randomUUID().toString();
+        Cookie csrfCookie = new Cookie(ApplicationConstants.CSRF_TOKEN,
```
[the remark this answers is not in the document]
Comment (platosha) [reply flow-server/src/main/java/com/vaadin/flow/server/VaadinServlet.java]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (platosha) [reply flow-server/src/main/java/com/vaadin/flow/server/VaadinServlet.java]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (platosha) [reply flow-server/src/main/java/com/vaadin/flow/server/VaadinServlet.java]: @Lift-dev ignore

**Review by haijian-vaadin — COMMENTED** [review]
The code looks good. But the Spring addon needs some change since it uses the `VaadinService.getCsrfTokenAttributeName()` API. 
No need to rush to v21, let's do it properly after vacation and make it to v22.

**Review by haijian-vaadin — APPROVED** [review]

**Pull request merged**